File: app/src/main/java/org/amoustakos/boilerplate/util/AndroidComponentUtil.java
package org.amoustakos.boilerplate.util;

import android.app.ActivityManager;
import android.app.ActivityManager.RunningServiceInfo;
import android.app.Service;
import android.content.ComponentName;
import android.content.Context;
import android.content.pm.PackageManager;


public final class AndroidComponentUtil {

    /**
     * Enable or disable a component (Activity, Service, BroadcastReceiver, or ContentProvider)
     */
    public static void setComponentState(Context context, Class componentClass, boolean enable) {
        ComponentName componentName = new ComponentName(context, componentClass);
        PackageManager pm = context.getPackageManager();
        pm.setComponentEnabledSetting(componentName,
                enable ? PackageManager.COMPONENT_ENABLED_STATE_ENABLED :
                        PackageManager.COMPONENT_ENABLED_STATE_DISABLED,
                PackageManager.DONT_KILL_APP);
    }

	/**
	 * Checks whether a service is running
	 */
	public static boolean isServiceRunning(Context context, Class<? extends Service> serviceClass) {
        ActivityManager manager =
                (ActivityManager) context.getSystemService(Context.ACTIVITY_SERVICE);

        for (RunningServiceInfo service : manager.getRunningServices(Integer.MAX_VALUE))
            if (serviceClass.getName().equals(service.service.getClassName()))
                return true;

        return false;
    }
}

File: app/src/main/java/org/amoustakos/boilerplate/util/AndroidComponentUtil.kt
package org.amoustakos.boilerplate.util

import android.content.ComponentName
import android.content.Context
import android.content.pm.PackageManager


object AndroidComponentUtil {

	/**
	 * Enable or disable a component (Activity, Service, BroadcastReceiver, or ContentProvider)
	 */
	fun setComponentState(context: Context, componentClass: Class<*>, enable: Boolean) {
		val componentName = ComponentName(context, componentClass)
		val pm = context.packageManager
		pm.setComponentEnabledSetting(componentName,
				if (enable)
					PackageManager.COMPONENT_ENABLED_STATE_ENABLED
				else
					PackageManager.COMPONENT_ENABLED_STATE_DISABLED,
				PackageManager.DONT_KILL_APP)
	}

	/*TODO: fix deprecation if possible
	 * Checks whether a service is running
	 */
//	fun isServiceRunning(context: Context, serviceClass: Class<out Service>): Boolean {
//		val manager = context.getSystemService(Context.ACTIVITY_SERVICE) as ActivityManager
//
//		manager.getRunningServices(Integer.MAX_VALUE).forEach { service ->
//			if (serviceClass.name == service.service.className)
//				return true
//		}
//
//		return false
//	}
}
